portal/code/company-portal/src/components/cax-header.tsx

```diff
@@ -14,50 +14,41 @@
 
 import * as React from 'react';
 import i18n from '../i18n';
-import { observer } from 'mobx-react';
-import { observable } from 'mobx';
 import { Row, Col } from 'react-bootstrap';
 import UserService from '../helpers/UserService';
-import { RouteComponentProps, withRouter } from 'react-router-dom';
 import { getUserClientRolesComposite } from '../helpers/utils';
+import { useState, useEffect } from "react";
+import { withRouter } from "react-router-dom";
 
-interface IProp extends RouteComponentProps {
-    href: string;
-    hidePivot?: boolean;
-    appTitle?: string;
-}
-
-@observer
-class Header extends React.Component<IProp> {
-    @observable username = '';
-    @observable initials = '';
-    @observable userRoles = [];
-    @observable language = i18n.language;
+export const Header = () => {
+    const username =  UserService.getUsername();
+    const initials =  UserService.getInitials();
+    const [userRoles, setuserRoles] =  useState([]);
+    const [language, setlanguage] =  useState(i18n.language);
 
-    public async componentDidMount() {
-        this.username = UserService.getUsername();
-        this.initials = UserService.getInitials();
-        this.userRoles = await getUserClientRolesComposite();
-    }
+    useEffect(() => {
+        // declare the data fetching function
+        const fetchData = async () => {
+          const data = await getUserClientRolesComposite();
+          setuserRoles(data);
+        }
+      
+        // call the function
+        fetchData()
+          // make sure to catch any error
+          .catch(console.error);
+      }, [])
 
-    private userClick() {
-        //const token = adalContext.getCachedToken();
-        const token = UserService.getToken();
-        console.log(token);
-    }
 
-    private logoutClick() {
-        const token = UserService.getCachedToken();
-        console.log(token);
-        UserService.logOut();
+    const logoutClick = () => {
+        // const token = UserService.getCachedToken();
```

Comment:
maybe remove those comments - actually it becomes a useless wrapper function. see my comment below